File: app/src/main/java/com/foxek/simpletimer/ui/timer/TimerContact.java
package com.foxek.simpletimer.ui.timer;

import com.foxek.simpletimer.ui.base.MvpInteractor;
import com.foxek.simpletimer.ui.base.MvpPresenter;
import com.foxek.simpletimer.ui.base.MvpView;

import io.reactivex.Flowable;
import io.reactivex.Observable;
import io.reactivex.Single;
import io.reactivex.disposables.Disposable;

public interface TimerContact {
    interface View extends MvpView {

        void startWorkoutActivity();

        void showPauseInterface();

        void showPlayInterface();

        void showCurrentCounter(String time);

        void showCounterType(int type);

        void showCounterNumber(String number);

        void showCounterName(String name);
    }

    interface Presenter extends MvpPresenter<View, Interactor> {

        void prepareIntervals(int workoutId);

        void pauseButtonClicked();

        void resetButtonClicked();
    }

    interface Interactor extends MvpInteractor {

        Flowable<Integer> fetchIntervalList(int workoutId);

        Observable<Integer> intervalFinishedCallback();

        Observable<Integer> tickCallback();

        Disposable getVolume(int id);

        boolean getTimerState();

        int getIntervalSize();

        void setTimerState(boolean timerState);

        void loadIntervalListToTimer();

        void continueTimer();

        void stopTimer();

        void indicateEndOfWorkout();

        void indicateEndOfInterval();

        void indicateLastSeconds();

        void deleteDependencies();

        String getIntervalName(int id);
    }
}

File: app/src/main/java/com/foxek/simpletimer/ui/timer/TimerContact.kt
package com.foxek.simpletimer.ui.timer

import com.foxek.simpletimer.data.model.Time
import com.foxek.simpletimer.data.timer.IntervalTimer

import io.reactivex.Observable
import io.reactivex.disposables.Disposable

interface TimerContact {

    interface ServiceCallback {

        fun startWorkoutActivity()

        fun showPauseInterface()

        fun showPlayInterface()

        fun showCurrentCounter(time: String)

        fun showCounterType(type: Int)

        fun showCounterNumber(number: String)

        fun showCounterName(name: String?)
    }

    interface Interactor {

        fun getTimerState(): IntervalTimer.State

        fun fetchIntervalList(workoutId: Int): Disposable

        fun intervalFinishedCallback(): Observable<Time>

        fun tickCallback(): Observable<Int>

        fun getVolume(id: Int): Disposable

        fun continueTimer()

        fun stopTimer()

        fun deleteDependencies()
    }
}